### PR TITLE
DHT republish

### PR DESCRIPTION
**Pull request by vitaliystoliarovcc**
- Republish in DHT mode
- Add start/end date options to `dht-read`

### CODE REVIEW

File: src/files-watcher.js [hunk 1]
```diff
@@ -51,29 +54,39 @@ class FilesWatcher extends EventEmitter {
...
+  }
 
+  /**
+   * @private
+   * @param {String} file
+   */
+  async watchFile (file) {
+    if (this.isReady) {
+      for await (const line of this.readFile(file)) {
+        this.emit('data', line) // republish new files
```
Comment (vigan-abd) [inline src/files-watcher.js:83]: should we emit also `file` here?

File: src/hyperswarm-dht-logger.js [hunk 2]
```diff
@@ -75,6 +133,7 @@ class HyperSwarmDHTLogger {
   }
 
   async stop () {
+    this.storage = []
```
Comment (vigan-abd) [inline src/hyperswarm-dht-logger.js:136]: is this used somewhere?

File: src/hyperswarm-dht-logger.js [hunk 3]
```diff
@@ -51,13 +73,49 @@ class HyperSwarmDHTLogger {
...
+          if (reachedEnd) {
+            reachedEnd = true
+            return false
+          } else if (date > startDate.getTime()) {
+            reachedEnd = true
+            return false
+          }
+        }
+      }
+      if (options.include && !data.match(options.include)) return false
```
Comment (vigan-abd) [inline src/hyperswarm-dht-logger.js:114]: did we setup these to on yargs?

**Review by vigan-abd — COMMENTED** [review]
added few minor comments

**Review by vigan-abd — APPROVED** [review]
lgtm

**Review by vigan-abd — CHANGES_REQUESTED** [review]
these tests are failing in my env, can we check them please?
```
 1) *** Unit testing! ***
       DHT logger tests
         logger should push data to the reader:

      AssertionError: expected +0 to equal 2
      + expected - actual

      -0
      +2
      
      at Context.<anonymous> (test/dht-logger.test.js:55:37)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  2) *** Unit testing! ***
       DHT logger tests
         filter logs by date:

      AssertionError: expected [] to deeply equal [ …(3) ]
      + expected - actual

      -[]
      +[
      +  "1970-01-01T00:10:00.000Z some data"
      +  "1970-01-01T00:20:00.000Z some data"
      +  "1970-01-01T00:30:00.000Z some data"
      +]
      
      at Context.<anonymous> (test/dht-logger.test.js:162:27)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  3) *** Unit testing! ***
       logger cli
         "after each" hook for "should print the logs":
     Error: Timeout of 20000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/vigan/Desktop/bitfinex/hypercore-logs/test/index.test.js)
      at listOnTimeout (internal/timers.js:557:17)
      at processTimers (internal/timers.js:500:7)
```

**Review by vigan-abd — COMMENTED** [review]
@vitaliystoliarovcc now getting tests failing with randon order
```
1) *** Unit testing! ***
       DHT logger tests
         filter logs by date - startDate option:

      AssertionError: expected [ …(5) ] to deeply equal [ …(4) ]
      + expected - actual

       [
         "1970-01-01T00:20:00.000Z some data"
      -  "1970-01-01T00:10:00.000Z some data"
         "1970-01-01T00:30:00.000Z some data"
         "1970-01-01T00:40:00.000Z some data"
         "1970-01-01T00:50:00.000Z some data"
       ]
      
      at Context.<anonymous> (test/dht-logger.test.js:233:27)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)

  2) *** Unit testing! ***
       DHT logger tests
         filter logs by date - endDate option:

      AssertionError: expected [ …(4) ] to deeply equal [ …(4) ]
      + expected - actual

       [
      +  "1970-01-01T00:00:00.000Z some data"
         "1970-01-01T00:10:00.000Z some data"
         "1970-01-01T00:20:00.000Z some data"
         "1970-01-01T00:30:00.000Z some data"
      -  "1970-01-01T00:00:00.000Z some data"
       ]
      
      at Context.<anonymous> (test/dht-logger.test.js:270:27)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

**Comment by vitaliystoliarovcc** [comment]
Ok, I'll try to increase delays

**Comment by vitaliystoliarovcc** [comment]
@vigan-abd done

**Review by vigan-abd — APPROVED** [review]

**Pull request merged**